pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -42,8 +42,17 @@ type ImageLocality struct {
 
 var _ framework.ScorePlugin = &ImageLocality{}
 
+var _ framework.PreScorePlugin = &ImageLocality{}
+
 // Name is the name of the plugin used in the plugin registry and configurations.
 const Name = names.ImageLocality
+// preScoreStateKey is the key in CycleState to ImageLocality pre-computed data for Scoring.
+const preScoreStateKey = "PreScore" + Name
+
+// preScoreState computed at PreScore and used at Score.
+type preScoreState struct {
+	containersPullAlways []v1.Container
```

Comment:
do we need a full list of containers? or just a single bool variable to indicate whether all containers are using PullAlways.

Reply:
I choose list of containers because I didn't want to loop over containers with pullPolicy set to Always twice. So ``preScoreState`` list is filled with containers on which ImageLocality score has to be calculated (other than Always).  I mean we can shrink it to a list of a strings ( just image property).